### PR TITLE
#10 - Adds wallet service and basic integration

### LINKED ISSUE #10: Integrate wallet service
## Description:

Integrate wallet service to: 
- Create a new wallet 
- Display balance of all assets 
- List the transactions

### PR DESCRIPTION
**Pull request by koshikraj**
## Description
Created wallet service with basic integration in creating wallet screen

- [x] Create a new wallet
- [x] Load a wallet account from an imported seed phrase
- [x] List basic details
- [x] List transactions

### CODE REVIEW

File: src/screens/create-wallet/create-wallet.screen.tsx [hunk 1]
```diff
@@ -31,12 +31,21 @@ export const CreateWalletScreen = observer(() => {
...
-      const safe = await safeService.create(walletBeneficiary, "test wallet", false);
+
+      const wallet = await walletService.create()
+      await walletService.info()
+
+      if(wallet.hasData()) {
+
+      const safe = await safeService.create(walletBeneficiary, wallet.data!.mnemonic, false);
       if (safe.hasData()) {
+        console.log(safe.data)
```
Comment (yathishram) [inline src/screens/create-wallet/create-wallet.screen.tsx]: We can remove the console.logs here

File: src/services/wallet/wallet.service.impl.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+     
+    } catch (e: any) {
+
+      return this.error<EthersWallet>(e.error)
+    }
+  }
+
+  // Service API to fetch the basic account info for a wallet
+
+  async info(): Promise<ServiceResponse<any>> {
```
Comment (yathishram) [inline src/services/wallet/wallet.service.impl.ts]: Is there a specific date type and not `any` for this?

File: src/store/account/account.store.impl.ts [hunk 3]
```diff
@@ -50,6 +50,7 @@ export class AccountStoreImpl extends StoreImpl implements AccountStore {
     });
     const injectedProvider = await web3Modal.connect();
     this.web3Provider = new Web3Provider(injectedProvider);
+    console.log(this.web3Provider.connection)
```
Comment (yathishram) [inline src/store/account/account.store.impl.ts]: We can remove this if not needed

File: src/utils/Wallet.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  address: string
+  mnemonic: string
+  wallet: EthersWallet
+}
+
+export type WalletInfo = {
+  address: string
+  balance: string
+  latestTransactions: any
+}
```
Comment (yathishram) [inline src/utils/Wallet.ts:25]: Can we add these types in a different folder so we can use them elsewhere also

File: src/utils/Wallet.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  }
+
+  async info(): Promise<ServiceResponse<any>> {
+
+    try {
+
+      const apiURL = NetworkUtil.getNetworkById(await this.walletProvider.getChainId())?.api
+      const address = await this.walletProvider.getAddress()
+      const balance = await this.walletProvider.getBalance()
+      const transactionsResp = await fetch(`${apiURL}&module=account&action=txlist&address=${address}&startblock=0&endblock=99999999&page=1&offset=5&sort=desc`).then(res => {return res.json()})
```
Comment (yathishram) [inline src/utils/Wallet.ts]: This is await right do we still need to do .`then`?

Comment (koshikraj) [reply src/utils/Wallet.ts]: Apparently, yes. `.json` returns a promise.

File: src/utils/Wallet.ts (hunk 4 above)
Comment (koshikraj) [reply src/utils/Wallet.ts:25]: We are maintaining types and enums on respective utils in frontened.

**Review by yathishram — APPROVED** [review]
LGTM

**Pull request merged**